### PR TITLE
README: update list of frameworks

### PR DESCRIPTION
**Pull request by jsquyres**
Signed-off-by: Jeff Squyres jsquyres@cisco.com

@hppritcha Please review
@hjelmn I took the liberty of adding "patcher" in the list, even though #1079 is not merged yet.

### CODE REVIEW

File: README [hunk 1]
```diff
@@ -1863,13 +1855,16 @@ ess       - RTE environment-specfic services
...
+notifier  - System-level notification support
 odls      - OpenRTE daemon local launch subsystem
 oob       - Out of band messaging
 plm       - Process lifecycle management
 ras       - Resource allocation system
 rmaps     - Resource mapping system
 rml       - RTE message layer
 routed    - Routing table for the RML
+rtc       - Run-time control framework
+schitzo   - OpenRTE personality framework
```
Comment (jjhursey) [inline README]: This should be "schizo" no 't'

Comment (jsquyres) [inline README]: Good catch; thanks -- will fix.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-pr/1548/ for details.

File: README [hunk 2]
```diff
@@ -1863,13 +1855,16 @@ ess       - RTE environment-specfic services
...
 sensor    - Software and hardware health monitoring
 snapc     - Snapshot coordination
```
Comment (hppritcha) [inline README]: snapc was  removed in v2.x

File: README [hunk 3]
```diff
@@ -1863,13 +1855,16 @@ ess       - RTE environment-specfic services
...
 sstore    - Distributed scalable storage
```
Comment (hppritcha) [inline README]: sstore was removed in v2.x

File: README [hunk 4]
```diff
@@ -1878,7 +1873,9 @@ state     - RTE state machine
 Miscellaneous frameworks:
 -------------------------
 
+allocator   - Memory allocator
 backtrace   - Debugging call stack backtrace support
+btl         - Point-to-point Byte Transfer Layer
 compress    - Compression algorithms
 crs         - Checkpoint and restart service
```
Comment (hppritcha) [inline README]: crs was removed in v2.x

Comment (jsquyres) [inline README]: Meh -- forgot to `git clean -dfx` between v1.10.x and v2.x checkout.  Fixed now.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-pr/1551/ for details.

File: README [hunk 5]
```diff
@@ -1863,25 +1855,25 @@ ess       - RTE environment-specfic services
...
 plm       - Process lifecycle management
 ras       - Resource allocation system
 rmaps     - Resource mapping system
 rml       - RTE message layer
 routed    - Routing table for the RML
-sensor    - Software and hardware health monitoring
-snapc     - Snapshot coordination
-sstore    - Distributed scalable storage
+rtc       - Run-time control framework
+schitzo   - OpenRTE personality framework
```
Comment (hppritcha) [inline README]: schizo

Comment (jsquyres) [inline README]: Arrgh.  Fixed.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-pr/1554/ for details.

**Comment by jsquyres** [comment]
@di0gen Please investigate this Jenkins failure -- this PR is a README change; there's no reason it should cause SIGABORT in the CI test.  Thanks.

**Comment by mike-dubman** [comment]
- orte oob and ud oob failing during multi-threaded test case
- also - `opal_mutex_unlock: Operation not permitted`
- @di0gen can help if it is jenkins CI infra related, which is not the case here.

``` shell
00:48:23 + taskset -c 16,17 timeout -s SIGSEGV 10m /var/lib/jenkins/jobs/gh-ompi-release-pr/workspace-2/ompi_install1/bin/mpirun -np 2 -bind-to core -mca btl_openib_if_include mlx4_0:1 -x MXM_RDMA_PORTS=mlx4_0:1 -x UCX_NET_DEVICES=mlx4_0:1 -x UCX_TLS=rc,cm -mca pml yalla /var/lib/jenkins/jobs/gh-ompi-release-pr/workspace-2/ompi_install1/thread_tests/thread-tests-1.1/latency_th 8
00:48:23 opal_mutex_unlock: Operation not permitted
00:48:23 [jenkins01:17237] *** Process received signal ***
00:48:23 [jenkins01:17237] Signal: Aborted (6)
00:48:23 [jenkins01:17237] Signal code:  (-6)
00:48:23 [jenkins01:17237] [ 0] /lib64/libpthread.so.0[0x3d6980f710]
00:48:23 [jenkins01:17237] [ 1] /lib64/libc.so.6(gsignal+0x35)[0x3d69032925]
00:48:23 [jenkins01:17237] [ 2] /lib64/libc.so.6(abort+0x175)[0x3d69034105]
00:48:23 [jenkins01:17237] [ 3] /var/lib/jenkins/jobs/gh-ompi-release-pr/workspace-2/ompi_install1/lib/openmpi/mca_oob_ud.so(+0x3efb)[0x7ffff4542efb]
00:48:23 [jenkins01:17237] [ 4] /var/lib/jenkins/jobs/gh-ompi-release-pr/workspace-2/ompi_install1/lib/openmpi/mca_oob_ud.so(+0x5f5b)[0x7ffff4544f5b]
00:48:23 [jenkins01:17237] [ 5] /var/lib/jenkins/jobs/gh-ompi-release-pr/workspace-2/ompi_install1/lib/libopen-rte.so.20(+0x78e54)[0x7ffff79dae54]
00:48:23 [jenkins01:17237] [ 6] /var/lib/jenkins/jobs/gh-ompi-release-pr/workspace-2/ompi_install1/lib/libopen-rte.so.20(orte_oob_base_set_addr+0x300)[0x7ffff79dab0d]
00:48:23 [jenkins01:17237] [ 7] /var/lib/jenkins/jobs/gh-ompi-release-pr/workspace-2/ompi_install1/lib/libopen-pal.so.20(opal_libevent2022_event_base_loop+0x53c)[0x7ffff76b6f2c]
00:48:23 [jenkins01:17237] [ 8] /var/lib/jenkins/jobs/gh-ompi-release-pr/workspace-2/ompi_install1/lib/libopen-pal.so.20(+0x398c7)[0x7ffff765a8c7]
00:48:23 [jenkins01:17237] [ 9] /lib64/libpthread.so.0[0x3d698079d1]
00:48:23 [jenkins01:17237] [10] /lib64/libc.so.6(clone+0x6d)[0x3d690e8b6d]
00:48:23 [jenkins01:17237] *** End of error message ***
00:48:23 --------------------------------------------------------------------------
00:48:23 mpirun noticed that process rank 0 with PID 0 on node jenkins01 exited on signal 6 (Aborted).
```

**Comment by rhc54** [comment]
FWIW: that's a Mellanox code area - can someone from Mellanox please address it?

@jladd-mlnx

**Comment by jsquyres** [comment]
@miked-mellanox @jladd-mlnx @Di0gen Following up on the mysterious Mellanox Jenkins error (which looks like it might be a larger race condition) here: https://github.com/open-mpi/ompi/issues/1586

**Comment by jladd-mlnx** [comment]
Thanks, Jeff! Looks like the bug was squashed. Do we know in which commit
this bug was introduced?

**Comment by jsquyres** [comment]
@jladd-mlnx See https://github.com/open-mpi/ompi/issues/1586#issuecomment-214898396

**Pull request merged**